### PR TITLE
[RISC-V] Add JIT Helpers for RISCV64

### PR DESCRIPTION
**Pull request by SzpejnaDawid**
I made analogously changes to https://github.com/dotnet/runtime/pull/103467
- JIT_GetDynamicNonGCStaticBase_SingleAppDomain
- JIT_GetDynamicGCStaticBase_SingleAppDomain

cc @dotnet/samsung. Part of https://github.com/dotnet/runtime/issues/84834.

### CODE REVIEW

**Comment by sirntar** [comment]
@SzpejnaDawid Please change title to `[RISC-V] Add JIT Helpers for RISCV64`, so our CI can test it.

**Comment by jkotas** [comment]
Is the fast path in these asm helpers better compared to what's generated by the C++ compiler?

**Comment by risc-vv** [comment]
<details><summary>e7506de4 is being scheduled for building and testing</summary>

GIT: `e7506de4aa4e65b0379674fc4035277707d5553a`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

File: src/coreclr/vm/riscv64/asmhelpers.S [hunk 1]
```diff
@@ -813,6 +813,39 @@ LEAF_END setFPReturn, _TEXT
 
 #endif // FEATURE_COMINTEROP
 
+// ------------------------------------------------------------------
+// void* JIT_GetDynamicNonGCStaticBase(DynamicStaticsInfo* pStaticsInfo)
+
+LEAF_ENTRY JIT_GetDynamicNonGCStaticBase_SingleAppDomain, _TEXT
+    // If class is not initialized, bail to C++ helper
+    ld a1, OFFSETOF__DynamicStaticsInfo__m_pNonGCStatics(a0)
+    nop
```
Comment (tomeksowi) [inline src/coreclr/vm/riscv64/asmhelpers.S:822]: What's the purpose of this nop?

Comment (SzpejnaDawid) [reply src/coreclr/vm/riscv64/asmhelpers.S]: Ah, I forgot to delete it. I fixed it

**Comment by risc-vv** [comment]
<details><summary>e85c158a is being scheduled for building and testing</summary>

GIT: `e85c158ae45ff5936bef6b9c941847988f26ee5b`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

**Comment by SzpejnaDawid** [comment]
> Is the fast path in these asm helpers better compared to what's generated by the C++ compiler?

I made these changes based on #99183 for ARM and #103467 for LOONGARCH. Before them, functions `JIT_GetDynamicNonGCStaticBase_Portable` and `JIT_GetDynamicGCStaticBase_SingleAppDomain` were called immediately. At now, we check `m_pNonGCStatics` and `m_pGCStatics` fields and then jump to the proper function. It cuts those calls when the proper field is `zero`. Routines `JIT_GetDynamicNonGCStaticBase_Portable` and `JIT_GetDynamicGCStaticBase_SingleAppDomain` are not leaves so additional frames come into play. This is also an overhead.

**Review by jkotas — APPROVED** [review]

**Pull request merged**